src/shogun/machine/Machine.h

```diff
@@ -328,11 +341,38 @@ class CMachine : public CStoppableSGObject
 		 */
 		virtual bool train_machine(CFeatures* data=NULL)
 		{
-			SG_ERROR("train_machine is not yet implemented for %s!\n",
-					get_name());
+			bool result=false;
+			
+			switch (data->get_feature_class())
```

Comment:
keep in mind data might be `nullptr`.
This actually raises a problem: as we cannot dispatch in that case ...
I guess we need to add a "default" option for training if features are not provided in the macro

Reply:
@shubham808 I think for now you can add a "train_without_features" that could be overloaded by subclasses. We can then remove that later

Reply:
default option will also work for features classes not yet in the switch cases, train_without_features  makes more sense